File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Energy Monitor Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var em = new Tinkerforge.BrickletEnergyMonitor(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current Energy Data
        em.getEnergyData(
            function (voltage, current, energy, realPower, apparentPower, reactivePower, powerFactor, frequecy) {
                console.log('Voltage: ' + voltage/100.0 + ' V');
                console.log('Current: ' + current/100.0 + ' A');
                console.log('Energy: ' + energy/100.0 + ' Wh');
                console.log('Real Power: ' + realPower/100.0 + ' h');
                console.log('Apparent Power: ' + apparentPower/100.0 + ' VA');
                console.log('Reactive Power: ' + reactivePower/100.0 + ' VAR');
                console.log('Power Factor: ' + powerFactor/1000.0);
                console.log('Frequency: ' + frequecy/100.0 + ' Hz');
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change XYZ to the UID of your Energy Monitor Bricklet

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_energy_monitor import BrickletEnergyMonitor

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    em = BrickletEnergyMonitor(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get current energy data
    voltage, current, energy, real_power, apparent_power, reactive_power, power_factor, \
      frequency = em.get_energy_data()

    print("Voltage: " + str(voltage/100.0) + " V")
    print("Current: " + str(current/100.0) + " A")
    print("Energy: " + str(energy/100.0) + " Wh")
    print("Real Power: " + str(real_power/100.0) + " h")
    print("Apparent Power: " + str(apparent_power/100.0) + " VA")
    print("Reactive Power: " + str(reactive_power/100.0) + " var")
    print("Power Factor: " + str(power_factor/1000.0))
    print("Frequency: " + str(frequency/100.0) + " Hz")

    input("Press key to exit\n") # Use raw_input() in Python 2
    ipcon.disconnect()
